src/test/java/org/apache/commons/io/FileUtilsTest.java

```diff
@@ -2463,18 +2485,56 @@ public void testReadFileToStringWithEncoding() throws Exception {
         assertEquals("Hello /u1234", data);
     }
 
+    @Test
+    public void testReadFileToString_Errors() {
+        assertThrows(NullPointerException.class, () -> FileUtils.readFileToString(null));
+        assertThrows(NoSuchFileException.class, () -> FileUtils.readFileToString(new File("non-exsistent")));
+
+        Exception exception = assertThrows(IOException.class, () -> FileUtils.readFileToString(tempDirFile));
+        assertEquals("Is a directory", exception.getMessage());
+
+        assertThrows(UnsupportedCharsetException.class, () -> FileUtils.readFileToString(tempDirFile, "unsupported-charset"));
+    }
+
+    @Test
+    @EnabledIf("isPosixFilePermissionsSupported")
+    public void testReadFileToString_AccessDeniedExceptionOnPosixFileSystem() throws Exception {
+        final File file = TestUtils.newFile(tempDirFile, "cant-read.txt");
+        TestUtils.createFile(file, 100);
+        Files.setPosixFilePermissions(file.toPath(), PosixFilePermissions.fromString("---------"));
+
+        assertThrows(AccessDeniedException.class, () -> FileUtils.readFileToString(file));
+    }
+
     @Test
     public void testReadLines() throws Exception {
         final File file = TestUtils.newFile(tempDirFile, "lines.txt");
-        try {
-            final String[] data = {"hello", "/u1234", "", "this is", "some text"};
-            TestUtils.createLineBasedFile(file, data);
+        final String[] data = {"hello", "/u1234", "", "this is", "some text"};
+        TestUtils.createLineBasedFile(file, data);
 
-            final List<String> lines = FileUtils.readLines(file, UTF_8);
-            assertEquals(Arrays.asList(data), lines);
-        } finally {
-            TestUtils.deleteFile(file);
```

Comment:
The file is created in the tempDir, so deleting it explicitly isn't necessary.

Reply:
NP, your repo, your decision ;-)

Do you want me to add the deletion in the `*_AccessDeniedExceptionOnPosixFileSystem`, too? I took my cue from the `testWriteString*` tests, where the cleanup relies on the tempDir.